### PR TITLE
multi-cloud connector in azure security center

### PR DESCRIPTION
**Pull request by aymanjarrousms**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [x] I have reviewed the [documentation](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/202/Overall-Process-of-AME-Onboarding) for the workflow.
- [x] [Validation tools](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/84/Swagger-Validation-tools) were run on swagger spec(s) and have all been fixed in this PR. [How to fix?](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/ci-fix.md)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/FAQ.md).

### ARM API Review Checklist
- [x] Service team MUST add the "**WaitForARMFeedback**" label if the management plane API changes fall into one of the below categories. 
  - adding/removing APIs.
  - adding/removing properties.
  - adding/removing API-version. 
  - adding a new service in Azure.

<i>Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.</i>

- [x] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process]( https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/212/Swagger-PR-Review).

### CODE REVIEW

**Review by chlahav — APPROVED** [review]

File: specification/security/resource-manager/Microsoft.Security/preview/2020-01-01-preview/connectors.json [hunk 1]
```diff
@@ -0,0 +1,618 @@
...
+                "description": "Install missing Azure Arc agent on VMs automatically"
+              },
+              {
+                "value": "Off",
+                "description": "Do not install Azure Arc agent on the VMs automatically"
+              }
+            ]
+          }
+        },
+        "resourceGroupName": {
```
Comment (ravbhatnagar) [inline specification/security/resource-manager/Microsoft.Security/preview/2020-01-01-preview/connectors.json:304]: which Resource group is this?

File: specification/security/resource-manager/Microsoft.Security/preview/2020-01-01-preview/connectors.json [hunk 2]
```diff
@@ -0,0 +1,618 @@
...
+    },
+    "ServicePrincipalProperties": {
+      "type": "object",
+      "description": "Details of the service principal.",
+      "properties": {
+        "applicationId": {
+          "type": "string",
+          "description": "Application id of service principal."
+        },
+        "secret": {
```
Comment (ravbhatnagar) [inline specification/security/resource-manager/Microsoft.Security/preview/2020-01-01-preview/connectors.json:339]: please ensure this value or any other secret value is not returned int he response of GET/PUT

File: specification/security/resource-manager/Microsoft.Security/preview/2020-01-01-preview/connectors.json [hunk 3]
```diff
@@ -0,0 +1,618 @@
...
+                "description": "Do not install Azure Arc agent on the VMs automatically"
+              }
+            ]
+          }
+        },
+        "resourceGroupName": {
+          "type": "string",
+          "description": "The name of the resource group."
+        },
+        "region": {
```
Comment (ravbhatnagar) [inline specification/security/resource-manager/Microsoft.Security/preview/2020-01-01-preview/connectors.json:308]: in this scenario, when customer is creating a connector to other clouds, what does region or resource group imply? which region is this?

**Review by ravbhatnagar — COMMENTED** [review]
@aymanjarrousms - few comments

File: specification/security/resource-manager/Microsoft.Security/preview/2020-01-01-preview/connectors.json (hunk 1 above)
Comment (aymanjarrousms) [reply specification/security/resource-manager/Microsoft.Security/preview/2020-01-01-preview/connectors.json:304]: In Azure Security Center multi-cloud scenario, we connect cloud VMs using hybrid compute (Arc) to Azure, where resourceGroupName is the resource group that the VMs connect to, and as you can see it's in the HybridComputeSettings

File: specification/security/resource-manager/Microsoft.Security/preview/2020-01-01-preview/connectors.json (hunk 2 above)
Comment (aymanjarrousms) [reply specification/security/resource-manager/Microsoft.Security/preview/2020-01-01-preview/connectors.json:339]: No secrets are returned, in the examples, these secrets are empty strings

File: specification/security/resource-manager/Microsoft.Security/preview/2020-01-01-preview/connectors.json (hunk 3 above)
Comment (aymanjarrousms) [reply specification/security/resource-manager/Microsoft.Security/preview/2020-01-01-preview/connectors.json:308]: As I wrote before regarding Azure Security Center scenario, we connect non-Azure Cloud VMs to Azure using Arc, these parameters are needed to onboard Arc agents,
- resource groups is used to connect the Arc to
- region is indication to the VM region

**Comment by raghushantha** [comment]
Approving changes [HybridRP Team]

**Comment by ravbhatnagar** [comment]
signing off from ARM side. generic connector will be explored later. Details in email.

**Review by zikalino — APPROVED** [review]

**Pull request merged**